x/pairing/types/key_provider_qos.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+import "strings"
+
+const ProviderQosStorePrefix = "ProviderQosStore/"
+
+func ProviderQosKey(provider string, chainID string, cluster string) string {
+	return strings.Join([]string{chainID, cluster, provider}, "/")
+}
+
+func GetProviderFromProviderQosKey(key string) string {
```

Comment:
I've scaffolded a new type: ProviderQos which keeps both the provider address and its qos in a single struct. Then, I changed the entries used in the providerQosFS to keep these objects instead of QoS objects only.
see https://github.com/lavanet/lava/commit/ba098997c73ca8ff9d2ba8f9f815d4428d74ceb7

Reply:
this means we double the storage for each one, lets think of another way to do this

Reply:
done. see https://github.com/lavanet/lava/commit/0ad34d2c5943afdf1a9d187e1a73bcbb3aab97df